analytics/saas_uploader.go

```diff
@@ -96,19 +99,23 @@ func (s *saasUploader) upload(tenant, fileName string) error {
 		return fmt.Errorf("http.NewRequest: %s", err)
 	}
 
-	req.Header.Set("Expect", "100-continue")
-	req.Header.Set("Content-Type", "application/x-gzip")
-	req.Header.Set("x-amz-server-side-encryption", "AES256")
+	if !s.isGCPManaged {
```

Comment:
A thought: Since inclusion of the headers are really part of the same decision as how to get the signedURL, what do you think about restructuring the code to have signedURLRequest() method instead of signedURL() and eliminating this condition here?

Reply:
This makes sense. As such s.isGCPManaged should only need to be checked once.

Reply:
Done. It should look cleaner and more readable now.